lib/db.js

```diff
@@ -302,6 +312,61 @@ class LimitDBRedis extends EventEmitter {
       });
   }
 
+  buildStatusResponse(bucket, prefixLength) {
+    return (key, lastDrip, remaining) => {
+      const bucketKeyConfig = this.bucketKeyConfig(bucket, key);
+      return {
+        key: key.substring(prefixLength),
+        limit: bucketKeyConfig.size,
+        reset: this.calculateReset(bucketKeyConfig, remaining),
+        remaining
+      };
+    };
+  }
+
+  /**
+   * Find all buckets that match the key.
+   *
+   * @param {statusParams} params
+   * @param {function(Error, statusResult)} [callback]
+   */
+  status(params, callback) {
```

Comment:
Maybe the right way to do this, even if it is not exactly a direct replacement is to return an stream and take advantage of redis scan operation which is already a cursor